queue_block.py

```diff
@@ -109,87 +92,78 @@ def pop(self, grp="null", count=1, reload=False):
             None
 
         '''
-        # determine which queue the signal is bound for
-        grp = self.group_by(signal)
-        grp = str(grp)
-        with self._queues_lock:
-            queue = self._queues[grp] = self._queues.get(grp, [])
+        queue = self._queues[grp]
 
-        # lock the queue before appending
-        lock = self._get_lock(grp)
-        with lock:
-            # check for uniqueness if property is set
-            try:
-                unique_val = self.uniqueness(signal)
-                self._logger.debug(
-                    "Testing uniqueness for signal: {}".format(unique_val)
-                )
-            except Exception as e:
-                unique_val = None
-                
-            if unique_val is not None:
-                for idx, sig in enumerate(self._queues[grp]):
-                    try:
-                        sig_val = self.uniqueness(sig)
-                    except Exception as e:
-                        sig_val = None
-                    if sig_val == unique_val:
-                        self._logger.debug(
-                            "Signal {} already in {}_queue".format(sig_val, grp)
-                        )
-                        if self.update:
-                            self._queues[grp][idx] = signal
-                        return
-
-            # pop one off the top of that queue if it's at capacity
-            if len(queue) == self.capacity:
-                self._logger.debug(
-                    "Pushing signal and capactity of {}_signal is full: {}".format(
-                        grp, self.capacity
+        # check for uniqueness if property is set
+        try:
+            unique_val = self.uniqueness(signal)
+            self._logger.debug(
+                "Testing uniqueness for signal: {}".format(unique_val)
+            )
+        except Exception as e:
+            unique_val = None
+
+        if unique_val is not None:
+            for idx, sig in enumerate(self._queues[grp]):
+                try:
+                    sig_val = self.uniqueness(sig)
+                except Exception as e:
+                    sig_val = None
+                if sig_val == unique_val:
+                    self._logger.debug(
+                        "Signal {} already in {}_queue".format(sig_val, grp)
                     )
+                    if self.update:
+                        self._queues[grp][idx] = signal
+                    return
+
+        # pop one off the top of that queue if it's at capacity
+        if len(queue) == self.capacity:
+            self._logger.debug(
+                "Pushing signal and capactity of {}_signal is full: {}".format(
+                    grp, self.capacity
                 )
-                # self.pop(grp) but without needing a new lock
-                if self.peek(grp):
-                    self._queues[grp] = self._queues[grp][1:]
+            )
+            self._queues[grp] = self._queues[grp][1:]
```

Comment:
Similar to before, should change this line.
Either:
```python
self._queues[grp][:] = self._queues[grp][1:]
```
or (my preference):
```python
self._queues[grp].pop(0)
```

I actually think this might even work and be more readable:
```python
queue.pop(0)
```